### ISSUE TITLE
Crash when logging a warning every second on iOS

### ISSUE THREAD

**Issue by troy-lamerton** [bugsnag/bugsnag-unity#243]
### Describe the bug

App crashes with stacktrace pointing to here:

```
BugsnagUnity.iOS.cpp:242:3
Client_Notify_mD1D951C54FCC258C8C5E62BD86419D0A9E9136CB
```

Bugsnag employees can [view the detailed error](https://app.bugsnag.com/the-naughty-cult-ltd/clash-of-streamers-1/errors/60985d3613c1a40008da10a1?filters%5Bevent.since%5D%5B%5D%5Btype%5D=eq&filters%5Bevent.since%5D%5B%5D%5Bvalue%5D=7d&filters%5Bapp.release_stage%5D%5B%5D%5Bvalue%5D=production&filters%5Bapp.release_stage%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Bvalue%5D=new&filters%5Bapp.type%5D%5B%5D%5Btype%5D=eq&filters%5Bapp.type%5D%5B%5D%5Bvalue%5D=iOS&event_id=60985d36007a7bf53c2d0000).

### Steps to reproduce

1. log a warning every second
2. do this several times on 100+ iOS devices
3. *crash event appears in Bugsnag dashboard*

### Environment
* Bugsnag version: v4.8.6
* Unity version: 2019.4.16f1
* iOS/Android/macOS/Windows/browser version: iOS, see [view the detailed error](https://app.bugsnag.com/the-naughty-cult-ltd/clash-of-streamers-1/errors/60985d3613c1a40008da10a1?filters%5Bevent.since%5D%5B%5D%5Btype%5D=eq&filters%5Bevent.since%5D%5B%5D%5Bvalue%5D=7d&filters%5Bapp.release_stage%5D%5B%5D%5Bvalue%5D=production&filters%5Bapp.release_stage%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Bvalue%5D=new&filters%5Bapp.type%5D%5B%5D%5Btype%5D=eq&filters%5Bapp.type%5D%5B%5D%5Bvalue%5D=iOS&event_id=60985d36007a7bf53c2d0000)
* simulator/emulator or physical device: see [view the detailed error](https://app.bugsnag.com/the-naughty-cult-ltd/clash-of-streamers-1/errors/60985d3613c1a40008da10a1?filters%5Bevent.since%5D%5B%5D%5Btype%5D=eq&filters%5Bevent.since%5D%5B%5D%5Bvalue%5D=7d&filters%5Bapp.release_stage%5D%5B%5D%5Bvalue%5D=production&filters%5Bapp.release_stage%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Btype%5D=eq&filters%5Berror.status%5D%5B%5D%5Bvalue%5D=new&filters%5Bapp.type%5D%5B%5D%5Btype%5D=eq&filters%5Bapp.type%5D%5B%5D%5Bvalue%5D=iOS&event_id=60985d36007a7bf53c2d0000)
* Initializing bugsnag via a Unity GameObject or in code?: code
* Player Settings:
    * Scripting backend (Mono or IL2CPP): IL2CPP
    * API compatibility level for .NET: .NET 4.x
    * Stack Trace level for all error types (None/ScriptOnly/Full): ScriptOnly

### Example code snippet

```cs
public static class TryReproduce {
    private static Timer _timer;

    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterSceneLoad)]
    public static void OnSceneLoaded() {
        _timer = new Timer((state) => {
            if (true) {
                Debug.LogWarning("the condition is true");
            }
        }, null, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(1));
    }
}
```

This code snippet above is my best guess at a reproduce - hasn't been confirmed to reproduce though.
We're not sure how to narrow down the issue further. Mention me if you need anything.

**Comment by johnkiely1** [comment]
Hi @troy-lamerton , 

I've been unable to reproduce this based on your code above, the error that get produced seems to indicate it is failing when trying to add breadcrumbs. However I believe it is more subtle and is very similar to https://github.com/bugsnag/bugsnag-cocoa/issues/822 in that it only occurs in a low memory situation. 

We have improved performance of breadcrumb logging in v6.2.3 of the `bugsnag-cocoa` (iOS) notifier to help reduce this type of occurrence. Although `bugsnag-unity` will use `bugsnag-cocoa` under the hood it is still using a v5 version of `bugsnag-cocoa` at the moment. We are in the process of updating the unity notifier, once complete so it should be able to make use of these performance improvement and reduce the likelihood of this happening.

I will update here when that upgrade has been completed.

**Comment by troy-lamerton** [comment]
Sounds like the issue is on our end. Improving memory usage is on our backlog todo.

@johnkiely1 is there anything more you need to do? Feel free to close this issue

**Comment by johnkiely1** [comment]
Hi @troy-lamerton, 

Just a quick update to say we have released [v5.0.0](https://github.com/bugsnag/bugsnag-unity/releases/tag/v5.0.0 ) of bugsnag-unity. It may be worth upgrading to that when you get a chance as it will have the improved breadcrumb logging performance.
